src/api/sign-collections.ts

```diff
@@ -0,0 +1,36 @@
+import { BaseAPI } from './base';
+
+interface SignNamespace {
+  signing_service?: string;
+  repository?: string;
+  namespace: string;
+}
+
+interface SignCollection extends SignNamespace {
+  collection?: string;
+}
+
+interface SignVersion extends SignCollection {
+  version?: string;
+}
+
+type SignProps = SignNamespace | SignCollection | SignVersion;
+
+class API extends BaseAPI {
+  UI_API_VERSION = 'v3';
+  apiPath = this.getUIPath('sign/collections/');
+
+  constructor() {
+    super(API_HOST + API_BASE_PATH);
+  }
+
+  getUIPath(url: string) {
+    return `${this.UI_API_VERSION}/${url}`;
+  }
```

Comment:
```suggestion
  apiPath = 'v3/sign/collections/';
```

(there is no such thing as a UI API v3, and as a bonus, the code gets simpler :))

Reply:
Oki, I started out like this but I am not certain why we need `API_HOST + API_BASE_PATH` in some urls and here why  we don't then?

Reply:
Well, we really use 3 different sets of APIs now:

* pulp apis .. `PulpApi` does `BaseApi('/pulp/api/v3/')` - this will become something like `API_HOST + PULP_API_BASE_PATH` (+-) when adding task management for cloud (#1589)
* hub apis .. `HubApi` does `BaseApi(API_HOST + API_BASE_PATH)`, `API_HOST` is usually empty, and `API_BASE_PATH` is either `/api/automation-hub/` or `/api/galaxy/`
  * UI apis .. are under hub, but use extra `getUIPath` to add `_ui/UI_API_VERSION/` between the base and `apiPath`

So.. you can drop the constructor as well, if you change from `extends BaseAPI` to `extends HubAPI` :)